src/metabase/upload.clj

```diff
@@ -339,6 +340,25 @@
          :num-rows          (count (rest rows))
          :generated-columns 0}))))
 
+(defn- create-from-csv-and-sync!
+  "This is separated from `create-csv-upload!` for testing"
```

Comment:
I extracted this because `create-csv-upload!` takes extra setup (e.g. perms) that we don't care about in most tests.

I struggled with naming here. We have three similarly named functions:
- `create-csv-upload!` (public entry point to this namespace, does everything including snowplow tracking)
- `create-from-csv-and-sync!` (creates a table in the customer DB and syncs it with our app DB)
- `create-from-csv!` (creates a table in the customer DB)